src/main/java/teammates/ui/webapi/action/GetStudentProfileAction.java

```diff
@@ -16,15 +18,53 @@ protected AuthType getMinAuthLevel() {
 
     @Override
     public void checkSpecificAccessControl() {
-        if (!userInfo.isStudent) {
-            throw new UnauthorizedAccessException("Student privilege is required to access this resource.");
+        if (userInfo.isStudent) {
+            checkStudentAccess();
+        } else if (userInfo.isInstructor) {
+            checkInstructorAccess();
```

Comment:
What if the account is a student of course A and an instructor of course B, in this case you will always treat it as student. Is there a way to differentiate? Probably adding some indicator param in the request?

The account should be able to view A's teammates's profile and profile of students in B(suppose it has privilege).

Reply:
Good catch!